inc/StyleUtils.php

```diff
@@ -100,6 +92,152 @@ public function cssStyleini($tpl, $preview=false) {
     }
 
 
+
+    /**
+     * Load style ini contents
+     *
+     * Loads and merges style.ini files from template and config and prepares
+     * the stylesheet modes
+     *
+     * @author Andreas Gohr <andi@splitbrain.org>
+     *
+     * @param string $tpl the used template
+     * @param bool   $preview load preview replacements
+     * @return array with keys 'stylesheets' and 'replacements'
+     */
+    public function cssStyleini($tpl, $preview=false) {
+        global $conf;
+
+        $mergedinis = $this->getStyleIni();
+
+        // allow replacement overwrites in preview mode
+        if($preview) {
+            $webbase = DOKU_BASE;
+            $ini     = $conf['cachedir'].'/preview.ini';
+            if(file_exists($ini)) {
+                $data = parse_ini_file($ini, true);
+                // FIXME does preview mode affect replacements only, or stylesheets too?
+                // replacements
+                if(is_array($data['replacements'])) {
+                    $mergedinis['replacements'] = array_merge(
+                        $mergedinis['replacements'],
+                        $this->cssFixreplacementurls($data['replacements'], $webbase)
+                    );
+                }
+            }
+        }
+
+        return array(
+            'stylesheets' => $mergedinis['stylesheets'],
+            'replacements' => $mergedinis['replacements']
+        );
+    }
+
+    protected function getStyleIni() {
+        static $inis = null;
+        if ( !$inis ) {
+            $inis = $this->retrieveStyles('styleini','parse_ini_file', array(true));
+
+            // FIXME is this trim filter still necessary?
+            foreach ($inis as $key => &$section) {
+                if ($section !== 'replacements') {
+                    continue;
+                }
+                $section = array_map('trim', $section);
+                $section = preg_replace('/^#.*/', '', $section);
+                $section = array_filter($section);
+            }
+        }
+        return $inis;
+    }
+
+    protected function retrieveStyles($type, $fn, $params = array()) {
```

Comment:
A doc block would help with reviewing here. Under what circumstances is `$type` not `styleini`? And when would `$fn` not be `parse_ini_file`?

Reply:
The useless parameters are removed.